src/server/set_family.cc

```diff
@@ -93,8 +93,25 @@ struct StringSetWrapper {
 
   unsigned Add(const NewEntries& entries, uint32_t ttl_sec) const {
     unsigned res = 0;
-    for (string_view member : EntriesRange(entries))
-      res += ss->Add(member, ttl_sec);
+    string_view members[StringSet::kMaxBatchLen];
+    bool res_arr[StringSet::kMaxBatchLen];
+    unsigned len = 0;
+    for (string_view member : EntriesRange(entries)) {
+      members[len++] = member;
+      if (len == StringSet::kMaxBatchLen) {
+        ss->AddMany(members, StringSet::kMaxBatchLen, ttl_sec, res_arr);
+        for (unsigned i = 0; i < StringSet::kMaxBatchLen; ++i)
+          res += res_arr[i];
```

Comment:
Hmmm, since we only use the `res to accumulate` why don't we do that in `StringSet::AddMany` ? So instead of returning an array of bool through res, just return directly the number of elements added?

You do this `accumulate` both here and a few lines below

Reply:
We also don't risk trigeerring ub, because if for some reason we call `AddMany` with `count < kMaxBatchLen` and try to use the `res` it won't be nice :smile:

Reply:
good idea. we can always change later.